torch/_tensor.py

```diff
@@ -1318,6 +1318,97 @@ def to_sparse_coo(self):
         """
         return self.to_sparse()
 
+    def dim_order(self):
+        """
+
+        dim_order() -> tuple
+
+        Returns a tuple of int describing the dim order or physical layout of :attr:`self`.
+
+        Args:
+            None
+
+        Dim order represents how dimensions are laid out in memory,
+        starting from the outermost to the innermost dimension.
+
+        Thus, the conversion from strides is done by sorting the strides
+        from larger to smaller since the dimension with the largest stride
+        is the outermost and the dimension with the smallest stride is the innermost.
+
+        For example, tensor with sizes = (3, 5, 2) and strides = (5, 1, 15), implies
+        physical layout of (2, 0, 1). Dimension order of (2, 0, 1) can be obtained
+        by sorting strides from large to smaller.
+
+        Example::
+            >>> torch.empty((2, 3, 5, 7)).dim_order()
+            (0, 1, 2, 3)
+            >>> torch.empty((2, 3, 5, 7), memory_format=torch.channels_last).dim_order()
+            (0, 2, 3, 1)
+
+        .. warning::
+            The dim_order tensor API is experimental and subject to change.
+
+        """
+
+        # NB:
+        # - Based on the implementation in TensorIterator.cpp
+        # - Should have similar behavior (esp in ambiguous cases) to,
+        #   torch._prims_common.compute_elementwise_output_logical_to_physical_perm
```

Comment:
Is there a reason we cannot literally share this code?

Reply:
Yeah I was thinking about doing the same at first. Then multiple tensors logic in the _prims_common function and given we are implementing a tensor method, I decided to copy and simplify.

I did add tests to make sure these impls are aligned, since that's what we want.

What do you think?

Reply:
You can just invoke that function with a single tensor?

Reply:
Yes, that is exactly what I am doing in the tests.

Reply:
Then why can we just not use that as actual impl, instead of copy paste?

Reply:
Alright, I will just call that function :)